monkey/infection_monkey/ransomware/ransomware_payload.py

```diff
@@ -67,3 +75,10 @@ def _add_extension(self, filepath: Path):
     def _send_telemetry(self, filepath: Path, error: str):
         encryption_attempt = RansomwareTelem((str(filepath), str(error)))
         self._telemetry_messenger.send_telemetry(encryption_attempt)
+
+    def _leave_readme(self):
+        if self._readme_enabled:
+            try:
+                shutil.copyfile(README_SRC, Path(self._target_dir) / README_DEST)
+            except Exception as ex:
+                LOG.warning(f"An error occurred while attempting to leave a README.txt file: {ex}")
```

Comment:
Why not move this and all readme-related code out of a payload and into an action?

Reply:
Is the readme part of the payload ?

Reply:
> Isn't the readme part of the payload ?

The readme is, but the the action of leaving a readme could potentially be something reusable.

> Why not move this and all readme-related code out of a payload and into an action?

We don't really have the infrastructure for actions defined, nor will we in this release. Regardless, I'm not sure if this is really a great candidate for an action. An action should be reusable and general. Leaving a readme is a specific case of "copy file". So the action we would build would be something like `CopyFileAction`, which you would call like `CopyFileAction(README_SRC, README_DEST, "An error message")` I'm not sure this would really gets us anything. In fact, it may make the action less flexible. Different users may want to log the error at a different log level.

IMHO, given the simplicity of this code, I'd prefer to leave it as is for now. If we see this pattern being repeated, then we can find a way to remove the duplication, which may or may not be an action.